modules/clightning/module.cpp

```diff
@@ -235,7 +245,44 @@ std::string clightning_des_offer(const std::string_view input) {
         result << hex_encode(compressed, 33);
     }
     
-    clean_tmpctx();
+    return result.str();
+}
+
+std::optional<std::string> clightning_parse_p2p_lightning_message(std::span<const uint8_t> buffer) {
+    CleanTmpCtxGuard _cleanup;
+
+    u8 *msg = (u8 *) tal_arr(tmpctx, u8, buffer.size());
+    memcpy(msg, buffer.data(), buffer.size());
+    peer_wire msg_type = (enum peer_wire)fromwire_peektype(msg);
+    std::ostringstream result;
+    
+    if (msg_type == WIRE_PING) {
+        u16 num_pong_bytes;
+	    u8 *ignored;
+        u8 *pong;
+
+        if (!check_ping_make_pong(tmpctx, msg, &pong)) {
+            return "";
+        }
+        if (!pong) {
+            return "";
+        }
+        fromwire_ping(tmpctx, msg, &num_pong_bytes, &ignored);
+
+        result << "MSG_TYPE=PING;NUM_PONG_BYTES=" << num_pong_bytes;
```

Comment:
Nit:  maybe it would be slightly more readable if the message types were lowercase, so they're distinguished more from the field names

```suggestion
        result << "MSG_TYPE=ping;NUM_PONG_BYTES=" << num_pong_bytes;
```

Reply:
Updated!